File: src/routes/translation.js
const express = require('express');
const router = express.Router();

/**
 * @route   GET /api/translation/languages
 * @desc    获取支持的语言列表
 * @access  Public
 */
router.get('/languages', async (req, res) => {
    try {
        const languages = [
            { code: 'zh-CN', name: '中文（简体）', flag: '🇨🇳' },
            { code: 'zh-TW', name: '中文（繁体）', flag: '🇹🇼' },
            { code: 'en', name: 'English', flag: '🇺🇸' },
            { code: 'ja', name: '日本語', flag: '🇯🇵' },
            { code: 'ko', name: '한국어', flag: '🇰🇷' },
            { code: 'fr', name: 'Français', flag: '🇫🇷' },
            { code: 'de', name: 'Deutsch', flag: '🇩🇪' },
            { code: 'es', name: 'Español', flag: '🇪🇸' }
        ];
        
        res.json({
            success: true,
            data: {
                languages: languages,
                total: languages.length
            }
        });
    } catch (error) {
        console.error('获取语言列表失败:', error);
        res.status(500).json({
            success: false,
            error: '获取语言列表失败'
        });
    }
});

/**
 * @route   POST /api/translation/translate
 * @desc    翻译文本
 * @access  Public
 */
router.post('/translate', async (req, res) => {
    try {
        const { text, fromLang, toLang } = req.body;
        
        if (!text || !fromLang || !toLang) {
            return res.status(400).json({
                success: false,
                error: '缺少必要参数'
            });
        }
        
        // 模拟翻译结果
        const mockTranslations = {
            'zh-CN': {
                'en': {
                    '你好': 'Hello',
                    '谢谢': 'Thank you',
                    '再见': 'Goodbye'
                }
            },
            'en': {
                'zh-CN': {
                    'Hello': '你好',
                    'Thank you': '谢谢',
                    'Goodbye': '再见'
                }
            }
        };
        
        let translatedText = text;
        if (mockTranslations[fromLang] && mockTranslations[fromLang][toLang]) {
            translatedText = mockTranslations[fromLang][toLang][text] || `[翻译] ${text}`;
        }
        
        const result = {
            originalText: text,
            translatedText: translatedText,
            fromLanguage: fromLang,
            toLanguage: toLang,
            confidence: 0.95,
            processingTime: Math.random() * 2 + 0.5
        };
        
        res.json({
            success: true,
            data: result
        });
    } catch (error) {
        console.error('翻译失败:', error);
        res.status(500).json({
            success: false,
            error: '翻译失败'
        });
    }
});

/**
 * @route   POST /api/translation/voice
 * @desc    语音翻译
 * @access  Public
 */
router.post('/voice', async (req, res) => {
    try {
        const { audioData, fromLang, toLang } = req.body;
        
        if (!audioData || !fromLang || !toLang) {
            return res.status(400).json({
                success: false,
                error: '缺少必要参数'
            });
        }
        
        // 模拟语音翻译结果
        const result = {
            originalText: '你好，很高兴认识你！',
            translatedText: 'Hello, nice to meet you!',
            originalAudio: audioData,
            translatedAudio: 'data:audio/wav;base64,mock_audio_data',
            fromLanguage: fromLang,
            toLanguage: toLang,
            confidence: 0.92,
            processingTime: 2.1
        };
        
        res.json({
            success: true,
            data: result
        });
    } catch (error) {
        console.error('语音翻译失败:', error);
        res.status(500).json({
            success: false,
            error: '语音翻译失败'
        });
    }
});

/**
 * @route   GET /api/translation/history
 * @desc    获取翻译历史
 * @access  Private
 */
router.get('/history', async (req, res) => {
    try {
        // 模拟翻译历史
        const mockHistory = [
            {
                id: 1,
                originalText: '你好',
                translatedText: 'Hello',
                fromLang: 'zh-CN',
                toLang: 'en',
                timestamp: new Date(Date.now() - 1000 * 60 * 30).toISOString(),
                type: 'text'
            },
            {
                id: 2,
                originalText: 'Thank you',
                translatedText: '谢谢',
                fromLang: 'en',
                toLang: 'zh-CN',
                timestamp: new Date(Date.now() - 1000 * 60 * 60).toISOString(),
                type: 'voice'
            }
        ];
        
        res.json({
            success: true,
            data: {
                history: mockHistory,
                total: mockHistory.length
            }
        });
    } catch (error) {
        console.error('获取翻译历史失败:', error);
        res.status(500).json({
            success: false,
            error: '获取翻译历史失败'
        });
    }
});

module.exports = router;

File: src/routes/translation.py
from flask import Blueprint, request, jsonify
import requests
import json

translation_bp = Blueprint('translation', __name__)

# 模拟翻译服务（实际项目中应该集成Google Translate API或其他翻译服务）
SUPPORTED_LANGUAGES = {
    'zh': '中文',
    'en': 'English',
    'ja': '日本語',
    'ko': '한국어',
    'fr': 'Français',
    'de': 'Deutsch',
    'es': 'Español',
    'it': 'Italiano',
    'pt': 'Português',
    'ru': 'Русский',
    'ar': 'العربية',
    'hi': 'हिन्दी',
    'th': 'ไทย',
    'vi': 'Tiếng Việt',
    'id': 'Bahasa Indonesia'
}

@translation_bp.route('/languages', methods=['GET'])
def get_supported_languages():
    """获取支持的语言列表"""
    return jsonify({
        'success': True,
        'languages': SUPPORTED_LANGUAGES,
        'total': len(SUPPORTED_LANGUAGES)
    })

@translation_bp.route('/text', methods=['POST'])
def translate_text():
    """文本翻译"""
    try:
        data = request.get_json()
        text = data.get('text', '')
        source_lang = data.get('source_lang', 'auto')
        target_lang = data.get('target_lang', 'en')
        
        if not text:
            return jsonify({
                'success': False,
                'error': 'Text is required'
            }), 400
        
        # 模拟翻译结果（实际项目中应该调用真实的翻译API）
        translated_text = f"[Translated from {source_lang} to {target_lang}] {text}"
        
        return jsonify({
            'success': True,
            'original_text': text,
            'translated_text': translated_text,
            'source_language': source_lang,
            'target_language': target_lang,
            'confidence': 0.95
        })
        
    except Exception as e:
        return jsonify({
            'success': False,
            'error': str(e)
        }), 500

@translation_bp.route('/voice', methods=['POST'])
def translate_voice():
    """语音翻译"""
    try:
        # 这里应该处理音频文件上传和语音识别
        # 目前返回模拟数据
        return jsonify({
            'success': True,
            'original_text': '你好，世界！',
            'translated_text': 'Hello, World!',
            'source_language': 'zh',
            'target_language': 'en',
            'audio_url': '/api/translation/audio/sample.mp3'
        })
        
    except Exception as e:
        return jsonify({
            'success': False,
            'error': str(e)
        }), 500

@translation_bp.route('/history', methods=['GET'])
def get_translation_history():
    """获取翻译历史"""
    # 模拟翻译历史数据
    history = [
        {
            'id': 1,
            'original_text': '你好',
            'translated_text': 'Hello',
            'source_lang': 'zh',
            'target_lang': 'en',
            'timestamp': '2025-06-25T10:00:00Z'
        },
        {
            'id': 2,
            'original_text': 'Thank you',
            'translated_text': '谢谢',
            'source_lang': 'en',
            'target_lang': 'zh',
            'timestamp': '2025-06-25T09:30:00Z'
        }
    ]
    
    return jsonify({
        'success': True,
        'history': history,
        'total': len(history)
    })

